### ISSUE TITLE
[1.13.0-rc1] Not possible to get DNS name for service tasks through "service ps"

### ISSUE THREAD

**Issue by thaJeztah** [docker/docker#28430]
From the discussion on https://github.com/docker/docker/pull/28088#issuecomment-260414020

> @mstanleyjones pointed out that because of the trunctation, it's no longer possible to use task IDs from service ps to find out what DNS name to use to connect to a specific task. There doesn't seem to be a straightforward way to do this. Should we reconsider the change?

We should either revert https://github.com/docker/docker/pull/28088 (but it may be required for https://github.com/docker/docker/issues/28089), or add a `--no-trunc` option (possibly as part of the formatting in https://github.com/docker/docker/pull/28213)

ping @aaronlehmann @aluzzardi @vieux we should decide on this before GA

**Comment by vieux** [comment]
as discussed with @aluzzardi , we are probably going to not truncate for `service ps` but still truncate for `logs`

**Comment by stevvooe** [comment]
How useful is the DNS name for a task? Not very and I am not sure that we should be doing more to encourage the usage of task ids directly.

The more annoying problem is using them for inspect. But this can be solved by supporting certain queries, rather than removing truncation.

**Comment by mstanleyjones** [comment]
@stevvooe if I want someone to run an example command on a service container (such as showing a secret using `docker exec` then I need a container ID to do it. I think we should eventually have `docker service exec` that will just pick a task container at random, but I don't think we have it yet.

**Comment by stevvooe** [comment]
I'm not sure what this has to do with DNS.

From docker 1.13-rc2:

```console
$ docker service ps myservice
NAME                      IMAGE  NODE  DESIRED STATE  CURRENT STATE          ERROR  PORTS
myservice.1.9ynjiorohirk  redis  moby  Running        Running 6 seconds ago
$ docker service ps myservice --no-trunc
NAME                      IMAGE  NODE  DESIRED STATE  CURRENT STATE           ERROR  PORTS
myservice.1.9ynjiorohirk  redis  moby  Running        Running 24 seconds ago
```

A few things:

1. `--no-trunc` is broken (I'm on rc2, so maybe it was fixed).
2. When truncating things, truncated names needed to be supported elsewhere. In this case, the truncated name should be usable in `docker inspect`, `docker exec`, `docker rm`, etc, since this maps to the prefix of a container name.

The problem is that we introduced truncation without introducing prefix lookup everywhere else a truncated name goes. Truncating isn't something you can just do. Please see the level of effort that goes into making this consistent in git. I know everyone is excited to make things pretty, but truncation needs to be thought through very carefully.

**Comment by aaronlehmann** [comment]
@aluzzardi: I think this has been fixed by #29196, right?
